dist/pythonlibs/testrunner/utils.py

```diff
@@ -20,8 +20,24 @@ def test_utils_interactive_sync(child, retries, delay):
         if ret == 0:
             break
     else:
-        # Last call to make it fail her,
+        # Last call to make it fail here,
         child.expect_exact('READY', timeout=0)
 
     child.sendline('s')
     child.expect_exact('START')
+
+
+def test_utils_interactive_sync_shell(child, retries, delay):
+    """Synchronisation for 'test_utils_interactive_sync' function
+    when shell module is used.
+
+    Interacts through input to wait for node being ready.
+    """
+    for _ in range(0, retries):
+        child.sendline('\n')
+        ret = child.expect_exact(['>', pexpect.TIMEOUT], timeout=delay)
+        if ret == 0:
+            break
+    else:
+        # Last call to make it fail here,
+        child.expect_exact('>', timeout=0)
```

Comment:
Both `test_utils_interactive_sync()` and `test_utils_interactive_sync_shell()` have the same code and just use some different values. How about generalizing the two? 

```py
def test_utils_interactive_sync(child, retries, delay, ready_cmd='r', ready_exp='READY'):
    """Synchronisation for 'test_utils_interactive_sync' function
    when shell module is used.
    Interacts through input to wait for node being ready.
    """
    for _ in range(0, retries):
        child.sendline(ready_cmd)
        ret = child.expect_exact([ready_exp, pexpect.TIMEOUT], timeout=delay)
        if ret == 0:
            break
    else:
        # Last call to make it fail here,
        child.expect_exact(ready_exp, timeout=0)


def test_utils_interactive_sync_shell(child, retries, delay):
    test_utils_interactive_sync(child, retries, delay, '\n', '>')
```

Reply:
Ah, just saw that they are not exactly the same, but still:

```py
def _test_utils_interactive_sync(child, retries, delay, ready_cmd='r', ready_exp='READY'):
    """Synchronisation for 'test_utils_interactive_sync' function
    when shell module is used.
    Interacts through input to wait for node being ready.
    """
    for _ in range(0, retries):
        child.sendline(ready_cmd)
        ret = child.expect_exact([ready_exp, pexpect.TIMEOUT], timeout=delay)
        if ret == 0:
            break
    else:
        # Last call to make it fail here,
        child.expect_exact(ready_exp, timeout=0)


def test_utils_interactive_sync(child, retries, delay):
    _test_utils_interactive_sync(child, retries, delay)
    child.sendline('s')
    child.expect_exact('START')


def test_utils_interactive_sync_shell(child, retries, delay):
    _test_utils_interactive_sync(child, retries, delay, '\n', '>')
```